wolfcrypt/src/rsa.c

```diff
@@ -1178,10 +1187,12 @@ static int RsaPublicEncryptEx(const byte* in, word32 inLen, byte* out,
 
     /* if async pending then return and skip done cleanup below */
     if (ret == WC_PENDING_E) {
+        key->state++;
```

Comment:
This is dangerous. The state values should not have to be monotonically increasing by one.
The alternative pattern is:
   ret = func();
   state = NEXT_STATE;
case NEXT_STATE:
    if (ret < 0)
       break;

Of course checking ret is redundant on re-entry ... but that is the tradeoff.